src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/upsertTranslation.hpp

```diff
@@ -0,0 +1,166 @@
+const char* INCLUDE_DIRECTORIES[] = {FLATBUFFER_SCHEMAS_DIR, nullptr};
+
+/**
+ * @brief UpsertTranslation class.
+ *
+ */
+class UpsertTranslation
+{
+private:
+    /**
+     * @brief Inserts a translation entry in a rocksdb database.
+     *
+     * @param data JSON with the required information to create or update a translation entry.
+     * @param rocksDbWrapper rocksdb wrapper instance.
+     */
+    static void insertTranslationEntry(const nlohmann::json& data, Utils::RocksDBWrapper& rocksDbWrapper)
+    {
+        std::string translationEntryStr;
+        if (rocksDbWrapper.get(data.at("resource").get<std::string>(), translationEntryStr))
+        {
+            std::cerr << "Key already exists." << std::endl;
+        }
+        else if (data.contains("payload"))
+        {
+            std::string flatbufferSchemaStr;
+            bool valid = (flatbuffers::LoadFile(FLATBUFFER_PATH.c_str(), false, &flatbufferSchemaStr));
+            if (!valid)
+            {
+                std::cerr << "Couldn't open flatbuffer schema: " << FLATBUFFER_PATH << std::endl;
+            }
+
+            const std::string payload = data.at("payload").dump();
+            const std::string resource = data.at("resource").get<std::string>();
+            flatbuffers::Parser parser;
+
+            valid = (parser.Parse(flatbufferSchemaStr.c_str(), INCLUDE_DIRECTORIES) && parser.Parse(payload.c_str()));
+            if (!valid)
+            {
+                std::cerr << "Error: " << parser.error_ << std::endl;
+            }
+
+            uint8_t* buf = parser.builder_.GetBufferPointer();
+            size_t flatbufferSize = parser.builder_.GetSize();
+
+            const rocksdb::Slice translationSlice(reinterpret_cast<const char*>(buf), flatbufferSize);
+            rocksDbWrapper.put(resource, translationSlice);
+        }
+        else
+        {
+            std::cerr << "No payload data" << std::endl;
+        }
+    }
+
+    /**
+     * @brief Updates a translation entry in a rocksdb database.
+     *
+     * @param data JSON with the required information to create or update a translation entry.
+     * @param rocksDbWrapper rocksdb wrapper instance.
+     */
+    static void updateTranslationEntry(const nlohmann::json& data, Utils::RocksDBWrapper& rocksDbWrapper)
+    {
+        std::string translationEntryStr;
+        if (rocksDbWrapper.get(data.at("resource").get<std::string>(), translationEntryStr))
+        {
+            const std::string resource = data.at("resource").get<std::string>();
+            std::string flatbufferSchemaStr;
+            flatbuffers::LoadFile(FLATBUFFER_PATH.c_str(), false, &flatbufferSchemaStr);
+            flatbuffers::Parser parser;
+            parser.opts.strict_json = true;
+
+            bool valid = parser.Parse(flatbufferSchemaStr.c_str(), INCLUDE_DIRECTORIES);
+            if (!valid)
+            {
+                std::cerr << parser.error_ << std::endl;
```

Comment:
Shouldn't this method return here if parsing was not valid?

Reply:
Fixed